graphql/mod.ts

```diff
@@ -0,0 +1,67 @@
+import { Drash } from "../deps.ts";
+import {
+  ExecutionResult,
+  graphql,
+  GraphQLSchema,
+  renderPlaygroundPage,
+} from "./deps.ts";
+
+interface GraphQLOptions {
+  schema: GraphQLSchema;
+  graphiql: boolean | string;
+  rootValue: Record<string, () => string>;
+}
+
+/**
+ * Taken from https://github.com/deno-libs/gql/blob/master/http.ts but heavily modified to suit drash's needs,
+ * and to utilise as much of graphql's own code instead
+ */
+export function GraphQL(
+  options: GraphQLOptions,
+): (
+  request: Drash.Http.Request,
+  response: Drash.Http.Response,
+) => Promise<Drash.Http.Response> {
+  return async (request: Drash.Http.Request, response: Drash.Http.Response) => {
+    const playgroundEndpoint = options.graphiql === true
+      ? "/graphql"
+      : typeof options.graphiql === "string"
+      ? options.graphiql
+      : undefined;
+    if (
+      playgroundEndpoint && request.method === "GET" &&
+      (request.headers.get("Accept")?.includes("text/html") ||
+        request.headers.get("Accept")?.includes("*/*"))
+    ) {
+      response.headers.set("Content-Type", "text/html");
+      response.body = renderPlaygroundPage({ endpoint: playgroundEndpoint });
+      return response;
+    }
+
+    if (!["PUT", "POST", "PATCH"].includes(request.method)) {
+      response.status_code = 405;
+      response.body = "Method Not Allowed";
+      return response;
+    }
+
+    const body = request.parsed_body.data as Record<string, unknown>;
+    const query = Object.keys(body)[0]; // Because drahs by default will parse body as application ww form url encoded, so the `body` looks like `{ "{ hello }": undefined }`, which is not the correct format and would need addressing ats ome point as drash should allow the body to be a string too
```

Comment:
wouldn't the query be in query params? is that what `parsed_body` is? i'm actually not sure. i imagine that the request would look like this: `dumb.net/graphql?query={hello}`

Reply:
HELLA typos LOL

Reply:
Well i've only seem people make a POST req to a graphql server (both in graphqls and gqls docs), so **at this moment in time**, this middleware is only for POST reqs, but guessing it should handle GETS too?

`parsed_body` is the body for POST (or PUT) reqs

Reply:
https://github.com/drashland/deno-drash-middleware/pull/118/commits/44818bae7ae45655df68d5d68bd42f71f09cf451 to address erics comment

Reply:
oh wait no you're right LOL `post` is definitely a thing. i think `get` is an old apollo way? only POST makes sense to keep in line with graphql docs